File: 019. Introducción a Spring WebFlux/spring-webflux/src/main/java/com/example/repository/BookRepository.java
package com.example.repository;

import com.example.model.Book;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

@Repository
public class BookRepository {

    private static Map<Long, Book> database;

    public BookRepository() {
        database = new ConcurrentHashMap<>();
        database.put(1L, new Book(1L, "book1", "author1"));
        database.put(2L, new Book(2L, "book2", "author2"));
        database.put(3L, new Book(3L, "book3", "author3"));
        database.put(4L, new Book(4L, "book4", "author4"));
        database.put(5L, new Book(5L, "book5", "author5"));
        database.put(6L, new Book(6L, "book6", "author6"));
        database.put(7L, new Book(7L, "book7", "author7"));
    }

    public Mono<Book> findById(Long id){
        return Mono.just(database.get(id));
    }

    public Flux<Book> findAll(){
        return Flux.fromIterable(database.values());
    }

    public Mono<Boolean> existById(Long id){
        return database.containsKey(id) ? Mono.just(true) : Mono.just(false);
    }

    public Mono<Book> save(Book book){
        if(book.getId() == null){
            Long id = database.values().stream().mapToLong(Book::getId).max().orElse(0L) + 1;
            book.setId(id);
        }
        database.put(book.getId(), book);
        return Mono.just(book);
    }

    public Mono<Void> deleteById(Long id){
        database.remove(id);
        return Mono.empty();
    }

    public Mono<Void> deleteAll(){
        database.clear();
        return Mono.empty();
    }

}

File: 020. Introducción a Spring WebFlux con Kotlin/spring-webflux-kotlin/src/main/kotlin/com/example/repository/BookRepository.kt
package com.example.repository

import com.example.domain.Book
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Repository
class BookRepository {

    private var database = mutableMapOf(
        1L to Book(1L, "Book1", "author 1"),
        2L to Book(2L, "Book2", "author 2"),
        3L to Book(3L, "Book3", "author 3"),
        4L to Book(4L, "Book4", "author 4"),
        5L to Book(5L, "Book5", "author 5"),
        6L to Book(6L, "Book6", "author 6"),
        7L to Book(7L, "Book7", "author 7"),
        8L to Book(8L, "Book8", "author 8"),
        9L to Book(9L, "Book9", "author 9"),
    )

    fun findById(id: Long): Mono<Book> = Mono.just(database[id]!!)

    fun findAll(): Flux<Book> = Flux.fromIterable(database.values)

    fun existById(id: Long): Mono<Boolean>{
        return if (database.containsKey(id)) Mono.just(true) else Mono.just(false)
    }

    fun save(book: Book): Mono<Book>{
        book.id = book.id ?: (database.values.mapNotNull { it.id }.maxOf { it.or(0L) } + 1)
        database[book.id!!] = book
        return Mono.just(book)
    }

    fun deleteById(id: Long): Mono<Void>{
        database.remove(id)
        return Mono.empty()
    }

    fun deleteAll(): Mono<Void>{
        database.clear()
        return Mono.empty()
    }

}